task/tkn-bundle/0.1/tkn-bundle.yaml

```diff
@@ -96,15 +100,42 @@ spec:
           done
         fi
       done
+      for f in "${FILES[@]}"; do
+        if [[ -n "$(params.STEPS_IMAGE)" ]]; then
+          yq e '(.spec.steps[] | select(has("image")).image) = "$(params.STEPS_IMAGE)"' -i $f
```

Comment:
It should probably have a way to only modify certain steps. But this is probably good enough for the first iteration. (Not required for this PR.)